src/transformers/models/mgp_str/modeling_mgp_str.py

```diff
@@ -0,0 +1,534 @@
+        elif isinstance(module, (nn.Linear, nn.Conv2d)):
+            module.weight.data = nn.init.trunc_normal_(module.weight.data, mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+    def _set_gradient_checkpointing(self, module: MGPSTREncoder, value: bool = False) -> None:
+        if isinstance(module, MGPSTREncoder):
+            module.gradient_checkpointing = value
+
+
+MGP_STR_START_DOCSTRING = r"""
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`MGPSTRConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+MGP_STR_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See [`ViTImageProcessor.__call__`]
+            for details.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_a3_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of a3 modules. See `a3_attentions` under returned tensors
+            for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    "The bare MGP-STR Model transformer outputting raw hidden-states without any specific head on top.",
+    MGP_STR_START_DOCSTRING,
+)
+class MGPSTRModel(MGPSTRPreTrainedModel):
+    def __init__(self, config: MGPSTRConfig):
+        super().__init__(config)
+        self.config = config
+        self.embeddings = MGPSTREmbeddings(config)
+        self.encoder = MGPSTREncoder(config)
+
+    def get_input_embeddings(self) -> nn.Module:
+        return self.embeddings.proj
+
+    @add_start_docstrings_to_model_forward(MGP_STR_INPUTS_DOCSTRING)
+    def forward(
+        self,
+        pixel_values,
+        output_attentions=None,
+        output_hidden_states=None,
+        return_dict=None,
+    ):
```

Comment:
nit: can be put on one line

```suggestion
    def forward(self, pixel_values, output_attentions=None, output_hidden_states=None, return_dict=None):
```